packages/medusa/src/services/inventory.ts

```diff
@@ -0,0 +1,93 @@
+import { BaseService } from "medusa-interfaces"
+import { MedusaError } from "medusa-core-utils"
+import { TransactionBaseService } from "../interfaces"
+import { EntityManager } from "typeorm"
+import ProductVariantService from "./product-variant"
+import { ProductVariant } from "../models"
+
+type InventoryServiceProps = {
+  manager: EntityManager
+  productVariantService: ProductVariantService
+}
+class InventoryService extends TransactionBaseService<InventoryService> {
+  protected productVariantService_: ProductVariantService
+
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  constructor({ manager, productVariantService }: InventoryServiceProps) {
+    // eslint-disable-next-line prefer-rest-params
+    super(arguments[0])
+
+    this.manager_ = manager
+    this.productVariantService_ = productVariantService
+  }
+
+  /**
+   * Updates the inventory of a variant based on a given adjustment.
+   * @param variantId - the id of the variant to update
+   * @param adjustment - the number to adjust the inventory quantity by
+   * @return resolves to the update result.
+   */
+  async adjustInventory(
+    variantId: string | undefined | null,
+    adjustment: number
+  ): Promise<ProductVariant | undefined> {
+    if (typeof variantId === "undefined" || variantId === null) {
+      return undefined
+    }
+
+    return this.atomicPhase_(async (manager) => {
+      const variant = await this.productVariantService_.retrieve(variantId)
+      // if inventory is managed then update
+      if (variant.manage_inventory) {
+        return await this.productVariantService_
+          .withTransaction(manager)
+          .update(variant, {
+            inventory_quantity: variant.inventory_quantity + adjustment,
+          })
+      } else {
+        return variant
+      }
+    })
+  }
+  /**
+   * Checks if the inventory of a variant can cover a given quantity. Will
+   * return true if the variant doesn't have managed inventory or if the variant
+   * allows backorders or if the inventory quantity is greater than `quantity`.
+   * @param variantId - the id of the variant to check
+   * @param quantity - the number of units to check availability for
+   * @return true if the inventory covers the quantity
+   */
+  async confirmInventory(
+    variantId: string,
+    quantity: number
+  ): Promise<boolean> {
+    // if variantId is undefined then confirm inventory as it
+    // is a custom item that is not managed
+    if (typeof variantId === "undefined" || variantId === null) {
```

Comment:
question: Same as previously, we should not be able to pass null or undefined for the inventory. I don't think that it makes sense. wdyt?

Reply:
For this, yes, as stated in the comment we should return true if it's a custom item

Reply:
But the consumer shouldn’t call that method if there is no variantId no? If the consumer has an item without variant id then it can directly return true without calling that method. Wdyt?

Reply:
What i mean is that, should we extract that condition to outside that method?

Reply:
Maybe keep for now and refactor at a later point; this way we stick to just converting in this PR. If we make the change you suggest we would also have to make changes everywhere this method is called :)

Reply:
@srindom true 👍